test/valid-data/type-mapped-enum-optional/schema.json

```diff
@@ -0,0 +1,30 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "definitions": {
+        "MyObject": {
+            "type": "object",
+            "properties": {
+                "a": {
+                    "type": "string",
+                    "enum": [
```

Comment:
Shouldn't the type we map to just be string?

Reply:
Yep good catch. Still trying to fully understand how everything works :)

Reply:
It's fun, right?